shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java

```diff
@@ -610,19 +622,41 @@ public ResponseEntity<ByteArrayResource> downloadStatistics(
 			@Parameter(description = "Subject name excluding regular expression", required=false) @Valid
 			@RequestParam(value = "subjectNameOutRegExp", required = false) String subjectNameOutRegExp
 			) throws RestServiceException, IOException {
+
+		ShanoirEvent event = null;
+		event = new ShanoirEvent(
+				ShanoirEventType.DOWNLOAD_STATISTICS_EVENT,
+				null,
+				KeycloakUtil.getTokenUserId(),
+				"Fetching statistics...",
+				ShanoirEvent.IN_PROGRESS,
+				null);
+
+		eventService.publishEvent(event);
+
+		createStats(studyNameInRegExp, studyNameOutRegExp, subjectNameInRegExp, subjectNameOutRegExp, event);
+
+		return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+	}
+
+	@Async
+	public void createStats(String studyNameInRegExp, String studyNameOutRegExp, String subjectNameInRegExp, String subjectNameOutRegExp, ShanoirEvent event) throws RestServiceException, IOException {
+		float progress = 0;
 		String tmpDir = System.getProperty(JAVA_IO_TMPDIR);
 		File userDir = DatasetFileUtils.getUserImportDir(tmpDir);
 		File statisticsFile = recreateFile(userDir + File.separator + "shanoirExportStatistics.tsv");
-		File zipFile = recreateFile(userDir + File.separator + "shanoirExportStatistics" + ZIP);
+		File zipFile = recreateFile(userDir + File.separator + "shanoirExportStatistics_" + event.getId() + ZIP);
 
 		// Get the data
-		try (	FileOutputStream fos = new FileOutputStream(statisticsFile);
-				BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos));	){
-
+		try (FileOutputStream fos = new FileOutputStream(statisticsFile);
+			BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos))){
 
 			List<Object[]> results = datasetService.queryStatistics(studyNameInRegExp, studyNameOutRegExp, subjectNameInRegExp, subjectNameOutRegExp);
 
 			for (Object[] or : results) {
+
+				progress += 1f / results.size();
+				event.setProgress(progress);
```

Comment:
If you want the progress to be updated "live", you need to publish it every time